table/table.go

```diff
@@ -101,108 +92,370 @@ func DefaultKeyMap() KeyMap {
 	}
 }
 
-// Styles contains style definitions for this list component. By default, these
-// values are generated by DefaultStyles.
+// Styles contains style definitions for this table component. Load default
+// styles to your table with [DefaultStyles].
 type Styles struct {
+	border       lipgloss.Border
+	borderStyle  lipgloss.Style
+	borderTop    bool
+	borderBottom bool
+	borderLeft   bool
+	borderRight  bool
+	borderColumn bool
+	borderHeader bool
+	borderRow    bool
+
 	Header   lipgloss.Style
 	Cell     lipgloss.Style
 	Selected lipgloss.Style
 }
 
-// DefaultStyles returns a set of default style definitions for this table.
+// DefaultStyles returns sensible default table styles.
 func DefaultStyles() Styles {
 	return Styles{
-		Selected: lipgloss.NewStyle().Bold(true).Foreground(lipgloss.Color("212")),
 		Header:   lipgloss.NewStyle().Bold(true).Padding(0, 1),
 		Cell:     lipgloss.NewStyle().Padding(0, 1),
+		Selected: lipgloss.NewStyle().Bold(true).Foreground(lipgloss.Color("212")).Padding(0, 1),
 	}
 }
 
-// SetStyles sets the table styles.
-func (m *Model) SetStyles(s Styles) {
-	m.styles = s
-	m.UpdateViewport()
+// NewFromTemplate lets you create a table [Model] from Lip Gloss'
+// [table.Table].
+func NewFromTemplate(t *table.Table, headers []string, rows [][]string) *Model {
+	m := &Model{
+		cursor:       0,
+		KeyMap:       DefaultKeyMap(),
+		Help:         help.New(),
+		table:        t,
+		useStyleFunc: true,
+	}
+	// We can't get the rows and headers from the table, so the user needs to
+	// provide them as arguments.
+	m.rows = rows
+	m.headers = headers
```

Comment:
FYI: I wrote a helper function on lipgloss to convert the internal `Data` into a `[][]string`:

https://github.com/charmbracelet/lipgloss/blob/c6b946e574dddb956f8642c9f84fa0219b777ed3/table/rows.go#L116

To avoid having to ask for the data again here, I think we should have a `GetHeaders()` and `GetData()` functions on lipgloss.

There is a small decision to make here. I'm not sure if `GetData()` should return a `[][]string` directly. For me, it makes more sense to make it return the internal `table.Data` interface and expose `dataToMatrix` as a public function.

Reply:
I agree, I think it makes more sense to only store the data once instead of managing the same set of values in two places (Model and lipgloss Table).

I vote that we add `GetData()` that returns a `Data` then the user can call `DataToMatrix`. Performance-wise it would be the same as we would probably be doing that type conversion under the hood if we were returning `[][]string`. This might lead users to want to include non-string values in the table... Is that something we want to support? I guess that feature could sit in the backlog for a bit while we think about it.